src/core/sr.ml

```diff
@@ -102,50 +87,109 @@ let check_rule : sym StrMap.t -> sym * pp_hint * rule Pos.loc -> unit =
     | Wild        -> assert false (* Cannot appear in LHS. *)
     | TRef(_)     -> assert false (* Cannot appear in LHS. *)
   in
-  let lhs = List.map (fun p -> Bindlib.unbox (to_m 0 p)) r.elt.lhs in
-  let lhs = Basics.add_args (Symb(s,h)) lhs in
-  let metas = Array.map (function Some m -> m | None -> assert false) metas in
-  (* We substitute the RHS with the corresponding metavariables. *)
-  let fn m =
-    let xs = Array.init m.meta_arity (Printf.sprintf "x%i") in
-    let xs = Bindlib.new_mvar mkfree xs in
-    let e = Array.map _Vari xs in
-    TE_Some(Bindlib.unbox (Bindlib.bind_mvar xs (_Meta m e)))
+  let lhs_args = List.map (fun p -> Bindlib.unbox (to_m 0 p)) rule.lhs in
+  let lhs = Basics.add_args (Symb(s,h)) lhs_args in
+  let lhs_metas =
+    Array.map (function Some m -> m | None -> assert false) lhs_metas
   in
-  let te_envs = Array.map fn metas in
-  let rhs = Bindlib.msubst r.elt.rhs te_envs in
-  (* Infer the type of the LHS and the constraints. *)
+
+  (* Build a map LHS pattern variable -> term_env variable. *)
+  let sym_name m = "&" ^ match m.meta_name with
+                         | Some(n) -> n
+                         | None    -> string_of_int m.meta_key
```

Comment:
Please do not use this style, move the concatenation into both branches and have the match on the next line.

Reply:
No, in this case, I find it better as it clearly shows that the resulting string always starts with '&'.

Reply:
In that case, at least change it to:
```ocaml
let sym_name m =
  let name =
    match ...
  in
  "&" ^ name
in
```
or
```ocaml
let sym_name m =
  "&" ^
  match ..
```